packages/flutter_tools/lib/src/commands/upgrade.dart

```diff
@@ -68,16 +69,22 @@ class UpgradeCommand extends FlutterCommand {
     printStatus('');
     printStatus(flutterVersion.toString());
 
-    final String projRoot = findProjectRoot();
-    if (projRoot != null) {
+    final String projectRoot = findProjectRoot();
+    if (projectRoot != null) {
       printStatus('');
-      await pubGet(context: PubContext.pubUpgrade, directory: projRoot, upgrade: true, checkLastModified: false);
+      await pubGet(context: PubContext.pubUpgrade, directory: projectRoot, upgrade: true, checkLastModified: false);
     }
 
     // Run a doctor check in case system requirements have changed.
     printStatus('');
     printStatus('Running flutter doctor...');
-    await doctor.diagnose();
+    code = await runCommandAndStreamOutput(
```

Comment:
Why does this need to be in a new process versus just invoking the doctor API?

Reply:
FYI, this also fixed an issue whereby running the doctor API from the version before we upgraded could crash the tool when run against an upgraded tree -- https://gist.github.com/tvolkert/c2872310667f019969d9990fcade1ebb